File: 0056-merge-intervals/0056-merge-intervals.cs
public class Solution{
  public int[][] Merge(int[][] intervals){
    //sort based on start.
    //if start  is overlaps with first intervals, merge.

    //intervals.Sort(based onstart),
    //result = {};
    //foreach(var interval in intervals){ if result is empty or not overlaps.
    // u add result.Add(interval) //new intevval
    //ow jsut updtte, merge whch updtate end index.
    Array.Sort(intervals, (a, b) => a[0].CompareTo(b[0]));

    List<int[]> result = new List<int[]>();

    foreach(int[] interval in intervals){
      if(result.Count == 0 || result.Last()[1] < interval[0]){
        //ur end of 2nd interval is smaler than first interval's start.
        result.Add(interval);
      }
      else{ // overlaps.
        // if( interval[1] < result.Last()[1])
        // {
        //   result.Last()[1] = result.Last()[1]
        // }
        result.Last()[1] = Math.Max(result.Last()[1], interval[1]);

      }
    }
    return result.ToArray();

  }
}

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
      intervals.sort(key= lambda x: x[0])

      result = []
      result.append(intervals[0])
      



      for i in range(1, len(intervals)):
        curr = intervals[i] # [2,6]
        last = result[-1]
        if last[1] >= curr[0]:
          #ie) 3>=2
          last[1] = max(last[1], curr[1])
        else:
          result.append(intervals[i])

      return result

          

